compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -346,19 +346,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     );
                 }
 
-                if let Some(def) = actual.ty_adt_def() {
-                    if let Some(full_sp) = tcx.hir().span_if_local(def.did()) {
-                        let def_sp = tcx.sess.source_map().guess_head_span(full_sp);
-                        err.span_label(
-                            def_sp,
-                            format!(
-                                "{} `{}` not found {}",
-                                item_kind,
-                                item_name,
-                                if def.is_enum() && !is_method { "here" } else { "for this" }
-                            ),
+                let span_and_label_suffix = match actual.kind() {
+                    ty::Param(param_type) => {
+                        let generics = self.tcx.generics_of(
+                            tcx.hir()
+                                .body_owner_def_id(hir::BodyId { hir_id: self.body_id })
+                                .to_def_id(),
```

Comment:
Sorry, I overcomplicated this. Instead of `tcx.hir().body_owner_def_id(..)`, let's just do `self.body_id.owner.to_def_id()` -- I've seen this being used elsewhere in typeck.